### DISCUSSION TITLE
Do something when the camera detect some object

### DISCUSSION THREAD

**Discussion by Haznul24** [ultralytics/yolov5#8952] [Q&A] [answered]
Hello, i want to ask about something, what should i do if i want to execute some lines when the camera detect some object?
For example, if the camera detect a person, the program will print "Hello Human", what should i add to the code
Thx

**Comment by ghost** [comment, marked as answer]
HI, I don‘t know if there is a better way but I would do something like this:
```
import torch
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')

while True:
    img = get picture from the camera
    results = model(img)   # detect objects and save results to a variable

    for result in results.pandas().xyxy[0]
        if result[6] == 'person': print('Hello Human')
```
You can see this in https://github.com/ultralytics/yolov5/issues/36

**Reply by Haznul24** [reply]
Hello, thanks for your answer. I have tried it, but after i add the code to detect.py there is an invalid syntax at 'get picture form the camera'. Is there any code should i add?

**Reply by ghost** [reply]
You should make a new file, place the code there and run this file instead of yolov5's detect.py. 'get a picture from the camera' is a placeholder for a method to take a photo. I would insert `from cv2 import VideoCapture`, after that `cam = VideoCapture(0)` and in the line where img is assigned `camResult, img = cam.read()`. Maybe you should learn a bit of python but I hope this helps you ^^

**Reply by Haznul24** [reply]
Hello, honestly I'm really new to Python and Yolov5. I have a project that uses detect.py and I want the program to do something like print or send data when the camera detects some object, but I still don't know how to do that. But thank you very much for the answer, it helps a lot for me to understand Yolov5.

**Reply by ghost** [reply]
Hi, you can try this. I think it should do what you have asked for:
```
import torch
from cv2 import VideoCapture

model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
cam = VideoCapture(0)

while True:
    camResult, img = cam.read()
    detections = model(img).pandas().xyxy[0]

    for detection in detections:
        if result[6] == 'person':
            print('Hello Human')
```
Instead of yolov5s you could replace it with another pretrained model like yolov5m which is a bit slower but better. You can see the different models on the main page of yolov5.
